src/cluster.jl

```diff
@@ -645,7 +659,12 @@ function create_worker(manager, wconfig)
         # require the value of config.connect_at which is set only upon connection completion
         for jw in PGRP.workers
             if (jw.id != 1) && (jw.id < w.id)
-                (jw.state === W_CREATED) && wait(jw.c_state)
+                # wait for wl to join
+                if jw.state === W_CREATED
+                    lock(jw.c_state) do
+                        wait(jw.c_state)
+                    end
+                end
```

Comment:
This seems potentially incorrect; shouldn't the equality comparison be within the `lock` block?

Reply:
Ah yes I think you're right, especially since the docstring for `c_state` says:
```julia
    c_state::Threads.Condition # wait for state changes, lock for state
```

Fixed in 9283e6f. But I see that `Worker.state` is an atomic, should we also be doing atomic reads of that variable everywhere else too? AFAICT most reads are currently non-atomic.

Reply:
We should probably switch to acquire-release, or keep everything seq-cst for now (reads without `@atomic` are `:monotonic`, so we should probably do reads as `@atomic :sequentially_consistent jw.state`.

Reply:
Acquire-release is probably more than strong enough for this.

Reply:
Alrighty, fixed in 8e35ba4. I stuck to seq-cst just because that's the default for `@atomic`.